core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -194,7 +194,7 @@ private void createServices() {
         if(Strings.isNullOrEmpty(servicesWithScalingSettings)) {
             runWithCompose("up " + flags);
         } else {
-            runWithCompose("up " + flags + " " + servicesWithScalingSettings);
+            runWithCompose("up " + flags + " " + servicesWithScalingSettings + " " + String.join(" ", this.services));
```

Comment:
uh? `servicesWithScalingSettings` already includes `services`: `Stream.concat(services.stream(), ...`

Reply:
I think there is a misunderstanding of how scaling works in docker-compose
if you want to scale some services you must specify their scaling exactly like what is done in servicesWithScalingSettings. For example, it would be something like:
docker-compose up --scale redis=2
the misunderstanding is here the previous command is do nothing different as the 
docker-compose up 
that brings up all services (only redis will have 2 instances)

But, if you want to bring up only some of the services (not all) one must specify the name of services. For example:
docker-compose up --scale redis=2 redis
this command will only brings up 2 instance of redis
or 
docker-compose up --scale redis=2 redis db
this command will set up 2 instance of redis and 1 instance of db

Reply:
Damn, I've been looking at this for ages and every time failing to understand the issue.

The current implementation _fails to limit the running services_. My misunderstanding was thinking that it was failing to launch any containers which are in `withServices` but not in `withScaledServices`.

Actually that is and always has been fine, but we're getting services launched that we do no want.

e.g. with a docker-compose.yml file like:
```
redis:
  image: redis
db:
  image: orchardup/mysql
  environment:
    MYSQL_DATABASE: wordpress
```

and

```
    public DockerComposeContainer environment = new DockerComposeContainer(new File("src/test/resources/compose-test.yml"))
        .withServices("redis");
```
(taken from the test)

... we'd expect on `redis` to come up, but actually `db` is launched to.

The same goes for:
```
    public DockerComposeContainer environment = new DockerComposeContainer(new File("src/test/resources/compose-test.yml"))
        .withServices("redis") // with or without this line
        .withScaledService("redis", 2);
```
... we get 2 `redis`, plus one `db` that wasn't asked for.

We should absolutely fix this.

@mnafshin thanks for the PR. I'd just like to add to the tests to ensure that we have coverage for:

* `withServices` is used (this is your existing test)
* `withServices` and `withScaledService` are both used together
* `withScaledService` is used on its own

If I can push to your branch I will do so.

Reply:
I'm glad I extended the test cases to be more comprehensive, because I've discovered some more ways this can break with different combinations of inputs. 

Slightly bigger commit incoming 😅

Reply:
It seems that I don't have permission to push to @mnafshin's fork, so I've created a new PR: #2922

Reply:
If you want I can add you, but I think it is a little late, and you have made your own repo